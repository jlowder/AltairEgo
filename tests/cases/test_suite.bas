5 LET SEED = RND(-1)
10 REM BASIC INTERPRETER TEST SUITE
20 REM Test basic arithmetic
30 PRINT "=== ARITHMETIC TESTS ==="
40 PRINT "2+3="; 2+3
50 PRINT "10-4="; 10-4
60 PRINT "5*6="; 5*6
70 PRINT "15/3="; 15/3
80 PRINT "2^3="; 2^3
90 PRINT "MOD: 7 MOD 3="; 7 MOD 3
100 REM Test variables
110 PRINT "=== VARIABLE TESTS ==="
120 A=5: B=10
130 PRINT "A="; A; " B="; B
140 C=A+B: PRINT "C=A+B="; C
150 REM Test string variables
160 A$="HELLO": B$="WORLD"
170 PRINT A$; " "; B$
180 REM Test arrays
190 PRINT "=== ARRAY TESTS ==="
200 DIM X(5), Y$(3)
210 FOR I=0 TO 5: X(I)=I*2: NEXT I
220 FOR I=0 TO 5: PRINT "X("; I; ")="; X(I): NEXT I
230 Y$(0)="FIRST": Y$(1)="SECOND": Y$(2)="THIRD"
240 FOR I=0 TO 2: PRINT "Y$("; I; ")="; Y$(I): NEXT I
250 REM Test FOR loops
260 PRINT "=== FOR LOOP TESTS ==="
270 FOR I=1 TO 5: PRINT I;: NEXT I: PRINT
280 FOR I=10 TO 2 STEP -2: PRINT I;: NEXT I: PRINT
290 REM Test nested loops
300 PRINT "=== NESTED LOOP TESTS ==="
310 FOR I=1 TO 3
320 FOR J=1 TO 2
330 PRINT I; ","; J; " ";
340 NEXT J
350 PRINT
360 NEXT I
370 REM Test IF statements
380 PRINT "=== IF STATEMENT TESTS ==="
390 A=5: B=10
400 IF A<B THEN PRINT "A is less than B"
410 IF A>B THEN PRINT "A is greater than B"
420 IF A=5 THEN PRINT "A equals 5"
430 IF A<>B THEN PRINT "A not equal to B"
440 REM Test GOSUB/RETURN
450 PRINT "=== SUBROUTINE TESTS ==="
460 GOSUB 1000
470 PRINT "Back from subroutine"
480 REM Test DATA/READ
490 PRINT "=== DATA/READ TESTS ==="
500 READ X, Y, Z$
510 PRINT "Read values: "; X; Y; Z$
520 REM Test functions
530 PRINT "=== FUNCTION TESTS ==="
540 PRINT "ABS(-5)="; ABS(-5)
550 PRINT "INT(3.7)="; INT(3.7)
560 PRINT "SQR(16)="; SQR(16)
570 PRINT "RND(1)="; RND(1)
580 PRINT "LEN(\"HELLO\")="; LEN("HELLO")
590 PRINT "LEFT$(\"HELLO\",2)="; LEFT$("HELLO",2)
600 PRINT "RIGHT$(\"HELLO\",2)="; RIGHT$("HELLO",2)
610 PRINT "MID$(\"HELLO\",2,2)="; MID$("HELLO",2,2)
620 REM Test TAB function
630 PRINT "=== TAB TESTS ==="
640 PRINT TAB(10); "TABBED TEXT"
650 PRINT "A"; TAB(5); "B"; TAB(10); "C"
660 END
1000 REM Subroutine
1010 PRINT "Inside subroutine"
1020 RETURN
2000 DATA 42, 3.14, "TEST STRING"
