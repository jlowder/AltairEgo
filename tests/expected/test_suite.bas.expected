Altair Ego: Emulating Altair BASIC 32K Rev. 3.2
OK
OK
SYNTAX ERROR
SYNTAX ERROR
SYNTAX ERROR
SYNTAX ERROR
ILLEGAL VARIABLE NAME
OK
